tests/src/tests/loop.spec.ts

```diff
@@ -366,4 +366,33 @@ export = () => {
 			expect(d).to.equal("o");
 		}
 	});
+
+	it("should support indexing tuple as array", () => {
+		const table = {
+			a:1,
+			b:2,
+			c:3,
+		}
+
+		for (const tuple of pairs(table)) {
+			expect(tuple[1]).to.equal(table[tuple[0]]);
+		}
+	});
+
+	it("should support iterator function with single return when indexing tuple as array", () => {
```

Comment:
This test and the one following don't actually do anything. Since the iterable function value is just `() => {}`, it never runs the inner loop, so nothing is tested.